### PR TITLE
[BugFix]log4j can not follow links

### PR DESCRIPTION
**Pull request by smartlxh**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

When the log directory is a soft link, expired log files cannot be deleted

image

image

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/common/Log4jConfig.java [hunk 1]
```diff
@@ -177,14 +177,6 @@ private static void reconfig() throws IOException {
         String sysRollNum = String.valueOf(Config.sys_log_roll_num);
         String sysDeleteAge = String.valueOf(Config.sys_log_delete_age);
 
-        if (!(sysLogLevel.equalsIgnoreCase("DEBUG") ||
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/common/Log4jConfig.java:180]: Why are you removing this?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by caneGuy — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by smartlxh** [comment]
backport 2.5.0

**Comment by smartlxh** [comment]
@Mergifyio backport 2.5.0

**Pull request merged**